packages/library/src/commonMain/kotlin/io/realm/internal/RealmObjectUtil.kt

```diff
@@ -84,15 +84,24 @@ fun <T : RealmObject> RealmObjectInternal.freeze(frozenRealm: RealmReference): T
  *
  * @param liveRealm Reference to the Live Realm that should own the thawed object.
  */
-internal fun <T : RealmObject> RealmObjectInternal.thaw(liveRealm: BaseRealm): T {
+internal fun <T : RealmObject> RealmObjectInternal.thaw(liveRealm: BaseRealm): T? {
     @Suppress("UNCHECKED_CAST")
     val type: KClass<T> = this::class as KClass<T>
     val managedModel = (`$realm$Mediator` as Mediator).createInstanceOf(type)
     val dbPointer = liveRealm.realmReference.dbPointer
-    return managedModel.manage(
-        liveRealm.realmReference,
-        `$realm$Mediator` as Mediator,
-        type,
-        RealmInterop.realm_object_thaw(`$realm$ObjectPointer`!!, dbPointer)
-    )
+    // FIXME C-API is currently throwing an error if the object has been deleted, so currently just
+    //  catching that and returning null
+    @Suppress("TooGenericExceptionCaught")
+    try {
+        return RealmInterop.realm_object_thaw(`$realm$ObjectPointer`!!, dbPointer)!!.let { thawedObject ->
+            managedModel.manage(
+                liveRealm.realmReference,
+                `$realm$Mediator` as Mediator,
+                type,
+                thawedObject
+            )
+        }
+    } catch (e: Exception) {
+        return null
```

Comment:
Should we check for a specific error message to not accidentally catch something else? Or is that just overkill since we are going to refactor this anyway?

Reply:
I added a guard on the exception test too.